File: chrome/test/data/extensions/bookmarkmanager/js/cr/event.js
// Copyright (c) 2010 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * @fileoverview This is a simple pure JS event class that can be used with
 * {@code cr.ui.EventTarget}. It should not be used with DOM EventTargets.
 */

cr.define('cr', function() {

  // cr.Event is called CustomEvent in here to prevent naming conflicts. We
  // alse store the original Event in case someone does a global alias of
  // cr.Event.

  const DomEvent = Event;

  /**
   * Creates a new event to be used with cr.EventTarget or DOM EventTarget
   * objects.
   * @param {string} type The name of the event.
   * @param {boolean=}
   * @constructor
   */
  function CustomEvent(type, opt_bubbles, opt_capture) {
    var e = cr.doc.createEvent('Event');
    e.initEvent(type, !!opt_bubbles, !!opt_capture);
    e.__proto__ = CustomEvent.prototype;
    return e;
  }

  CustomEvent.prototype = {
    __proto__: DomEvent.prototype
  };

  // Export
  return {
    Event: CustomEvent
  };
});

File: aura/event.cc
// Copyright (c) 2011 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "aura/event.h"

namespace aura {

Event::Event(ui::EventType type, int flags)
    : type_(type),
      time_stamp_(base::Time::NowFromSystemTime()),
      flags_(flags) {
  Init();
}

Event::Event(NativeEvent native_event, ui::EventType type, int flags)
    : type_(type),
      time_stamp_(base::Time::NowFromSystemTime()),
      flags_(flags) {
  InitWithNativeEvent(native_event);
}

Event::Event(const Event& copy)
    : native_event_(copy.native_event_),
      type_(copy.type_),
      time_stamp_(copy.time_stamp_),
      flags_(copy.flags_) {
}

}  // namespace aura

